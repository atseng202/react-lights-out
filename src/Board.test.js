import React from "react";

import { render, fireEvent } from "@testing-library/react";
import Board from "./Board";

/* ******************** Smoke Tests  ******************************* */
describe("smoke tests for Board", function () {
  it("renders without crashing", function () {

    render(<Board />);
  });
  // end
});

/* ******************** Snapshot Tests  ******************************* */

describe("snapshot for individual Board", function () {
  it("renders accurate board from provided props", function () {

    const { container, debug } = render(
      <Board 
        chanceLightStartsOn={0}
      />
    );
  
    expect(container).not.toContainHTML("Cell-lit");
    expect(container).toMatchSnapshot();
  });
  // end
});

/* ******************** Cell Click Tests  ******************************* */

describe("cell clicking", function () {
  it("correctly toggles the cells as lit/not lit (all sides + clicked cell)", function () {

    const { container, debug } = render(
      <Board 
        nrows={3}
        ncols={3}
        chanceLightStartsOn={1}
      />
    );
  
    fireEvent.click(container.querySelectorAll("tr")[1].querySelectorAll("td")[1]);
    expect(container.querySelectorAll("tr")[1].querySelectorAll("td")[1]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[0].querySelectorAll("td")[1]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[2].querySelectorAll("td")[1]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[1].querySelectorAll("td")[0]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[1].querySelectorAll("td")[2]).not.toContainHTML("Cell-lit");      
    
    // expect(container).not.toContainHTML("Cell-lit");
    // expect(container).toMatchSnapshot();
  });

  it("correctly toggles the cells as lit/not lit (top left corner)", function () {

    const { container, debug } = render(
      <Board 
        nrows={3}
        ncols={3}
        chanceLightStartsOn={1}
      />
    );
  
    fireEvent.click(container.querySelectorAll("tr")[0].querySelectorAll("td")[0]);
    expect(container.querySelectorAll("tr")[0].querySelectorAll("td")[0]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[1].querySelectorAll("td")[0]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[0].querySelectorAll("td")[1]).not.toContainHTML("Cell-lit");      
    
    // expect(container).not.toContainHTML("Cell-lit");
    // expect(container).toMatchSnapshot();
  });
  
  it("correctly toggles the cells as lit/not lit (bottom right corner)", function () {

    const { container, debug } = render(
      <Board 
        nrows={3}
        ncols={3}
        chanceLightStartsOn={1}
      />
    );
  
    fireEvent.click(container.querySelectorAll("tr")[2].querySelectorAll("td")[2]);
    expect(container.querySelectorAll("tr")[2].querySelectorAll("td")[2]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[2].querySelectorAll("td")[1]).not.toContainHTML("Cell-lit");      
    expect(container.querySelectorAll("tr")[1].querySelectorAll("td")[2]).not.toContainHTML("Cell-lit");      
    
    // expect(container).not.toContainHTML("Cell-lit");
    // expect(container).toMatchSnapshot();
  });
  // end
});

/* ******************** End of Game Tests  ******************************* */

describe("end of game", function () {
  it("win message properly displays when no lit cells", function () {

    const { container, debug } = render(
      <Board 
        chanceLightStartsOn={0}
      />
    );
  
    expect(container.querySelector(".Board-winMsg")).toContainHTML("Winner");
  });
  
  it("no win message there are lit cells", function () {

    const { container, debug } = render(
      <Board 
        chanceLightStartsOn={1}
      />
    );
  
    expect(container.querySelector(".Board-winMsg")).toBe(null);
  });
  // end
});


