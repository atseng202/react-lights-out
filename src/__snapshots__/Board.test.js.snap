// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for individual Board renders accurate board from provided props 1`] = `
<div>
  <div
    class="Board-winMsg"
  >
    Winner!
  </div>
</div>
`;
